scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/SubmitWindow.java

```diff
@@ -1260,12 +1208,72 @@ private void setEnabledStartAtPart(boolean state) {
         }
     }
 
-    private void build() {
+    private void uploadJobXMLForResubmit() {
+        RequestBuilder req = new RequestBuilder(RequestBuilder.GET, URL_UPLOAD_FILE);
+        req.setHeader(SESSION_ID_PARAMETER_NAME, LoginModel.getInstance().getSessionId());
+        req.setHeader(JOB_ID_PARAMETER_NAME, jobId);
+        req.setCallback(new RequestCallback() {
+            @Override
+            public void onResponseReceived(Request request, Response response) {
+                handleUploadDoneResponse(response.getText());
+            }
 
-        buildCatalogUrl();
+            @Override
+            public void onError(Request request, Throwable exception) {
+                // Exception is handled on request.send()
+            }
+        });
+        try {
+            req.send();
+        } catch (RequestException e) {
+            GWT.log("Error occurred when fetching workflow xml for job " + jobId);
+            // For Dev debug mode only
+            e.printStackTrace();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)